### ISSUE TITLE
Parser should ignore { CREATE | DROP } EXTENSION statement

### ISSUE THREAD

**Issue by gsdatta** [jOOQ/jOOQ#8324]
I have a flyway migration for a postgres database being parsed with the following:
```sql
CREATE EXTENSION pgcrypto;
```

I'm getting the following error when I try to use the DDL code generation feature:
```
org.jooq.impl.ParserException: GENERATOR, GLOBAL TEMPORARY TABLE, INDEX, OR ALTER VIEW, OR REPLACE VIEW, SCHEMA, SEQUENCE, TABLE, TEMPORARY TABLE, UNIQUE INDEX, or VIEW expected: [1:8] CREATE [*]EXTENSION pgcrypto;
```

Is there a way to configure the DDL parser to ignore this migration or handle it correctly?

Thanks!

**Comment by lukaseder** [comment]
Thanks for your request. `CREATE EXTENSION` statements are currently not supported by the jOOQ parser. As they're generally not too interesting to anything jOOQ does, I'm not sure if we should prioritise their support. However, perhaps it should be possible for users to "comment" certain vendor-specific statements with a jOOQ-specific comment syntax. This way, your `CREATE EXTENSION` statement will run just fine in PostgreSQL, but will be ignored by the jOOQ parser. I've created a feature request for this, which will be implemented in jOOQ 3.12: #8325.

I can't think of any workarounds right now, short of preprocessing your DDL files and stripping the statement(s) manually e.g. using regular expressions prior to passing the files to jOOQ.

**Comment by gsdatta** [comment]
Makes sense, thanks for the response. I've moved it to an init script for now.

On a related note, one of my DDLs uses a function (pgcrypto's `gen_random_uuid()`) as default value. The parser also fails on this with "Unknown function" - any way to tell the parser about this function?

I tried creating a function with `CREATE FUNCTION` to wrap the pgcrypto function, but H2 doesn't support PG functions so that didn't work.

**Comment by lukaseder** [comment]
> On a related note, one of my DDLs uses a function (pgcrypto's `gen_random_uuid()`) as default value. The parser also fails on this with "Unknown function" - any way to tell the parser about this function?

Will be happy to look into this. Can you create a new issue for this with some more details on how to reproduce it?

**Comment by lukaseder** [comment]
I'll increase the priority for this to be included in jOOQ 3.12, as it seems to be a very recurring issue when using pgdump exports with the `DDLDatabase`. Unfortunately, pgdump always produces these statements, even if they are usually not needed.

We'll support `CREATE EXTENSION` (https://www.postgresql.org/docs/current/sql-createextension.html) and `DROP EXTENSION` (https://www.postgresql.org/docs/current/sql-dropextension.html). `ALTER EXTENSION` will be out of scope for now.

**Comment by lukaseder** [comment]
This statement will now be ignored in jOOQ 3.12
